hoomd/md/pytest/test_thermo.py

```diff
@@ -0,0 +1,166 @@
+import hoomd
+import numpy as np
+
+""" Each entry is a quantity and its type """
+_thermo_qtys = [
+    ('kinetic_temperature', float),
+    ('pressure', float),
+    ('pressure_tensor', list),
+    ('kinetic_energy', float),
+    ('translational_kinetic_energy', float),
+    ('rotational_kinetic_energy', float),
+    ('potential_energy', float),
+    ('degrees_of_freedom', int),
+    ('translational_degrees_of_freedom', int),
+    ('rotational_degrees_of_freedom', int),
+    ('num_particles', int),
+]
+
+def test_attach_detach(simulation_factory, two_particle_snapshot_factory):
+    # test before attaching to simulation
+    group = hoomd.filter.All()
+    thermo = hoomd.md.compute.ThermodynamicQuantities(group)
+    for qty, typ in _thermo_qtys:
+        assert getattr(thermo, qty) is None
+
+    # make simulation and test state of operations
+    sim = simulation_factory(two_particle_snapshot_factory())
+    sim.operations.add(thermo)
+    assert len(sim.operations.computes) == 1
+    sim.operations.schedule()
+
+    # make sure quantities are computable without failure
+    for qty, typ in _thermo_qtys:
+        calc_qty = getattr(thermo, qty)
+        assert type(calc_qty) == typ
+
+    # detach from simulation and test properties again
+    sim.operations.remove(thermo)
+    assert len(sim.operations.computes) == 0
+    for qty, typ in _thermo_qtys:
+        assert getattr(thermo, qty) is None
+
+def test_basic_system_3d(simulation_factory, two_particle_snapshot_factory):
+    group = hoomd.filter.All()
+    thermo = hoomd.md.compute.ThermodynamicQuantities(group)
+    snap = two_particle_snapshot_factory()
+    if snap.exists:
+        snap.particles.velocity[:] = [[-2, 0, 0], [2, 0, 0]]
+    sim = simulation_factory(snap)
+    sim.always_compute_pressure = True
+    sim.operations.add(thermo)
+
+    integrator = hoomd.md.Integrator(dt=0.0001)
+    integrator.methods.append(hoomd.md.methods.NVT(group, tau=1, kT=1))
+    sim.operations.integrator = integrator
+
+    sim.operations.schedule()
+    sim.run(1)
+
+    assert thermo.num_particles == 2
+    assert thermo.rotational_degrees_of_freedom == 0
+    assert thermo.translational_degrees_of_freedom == 3
+    assert thermo.degrees_of_freedom == 3
+
+    np.testing.assert_allclose(thermo.potential_energy, 0.0)
+    np.testing.assert_allclose(thermo.rotational_kinetic_energy, 0.0, rtol=1e-5)
+    np.testing.assert_allclose(thermo.translational_kinetic_energy, 4.0, rtol=1e-5)
+    np.testing.assert_allclose(thermo.kinetic_energy, 4.0, rtol=1e-5)
+    np.testing.assert_allclose(thermo.kinetic_temperature, 2*thermo.kinetic_energy/thermo.degrees_of_freedom, rtol=1e-5)
+    np.testing.assert_allclose(thermo.pressure, 2.0/3*thermo.kinetic_energy/20**3, rtol=1e-5)
+    np.testing.assert_allclose(thermo.pressure_tensor, [8.0/20.0**3, 0., 0., 0., 0., 0.], rtol=1e-5)
+
+
+def test_basic_system_2d(simulation_factory, lattice_snapshot_factory):
+    group = hoomd.filter.Type(['A'])
```

Comment:
If we are going to name the other filter `groupB` this should be `groupA` although I prefer the names `filter_A` and `filter_B`.